### PR TITLE
Reject unaccepted pointers in Drag recognizer

### LINKED ISSUE #75061: '_initialButtons == kPrimaryButton': is not true.
Corresponds to Google internal bug b/179634050

1. Run `flutter create bug`.
2. Update the files as follows: 
```
import 'package:flutter/material.dart';

class StartPage extends StatefulWidget {
  @override
  _StartPageState createState() => _StartPageState();
}

class _StartPageState extends State<StartPage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(
          'Start',
          style: TextStyle(fontSize: 30, fontWeight: FontWeight.w700),
        ),
        centerTitle: false,
        backgroundColor: Color(0xff06060c),
      ),
      backgroundColor: Color(0xff06060c),
      body: Column(
        children: [
          Expanded(
            child: ListView(
              children: [
                Container(
                  decoration:
                      BoxDecoration(borderRadius: BorderRadius.circular(10)),
                  height: 170,
                  child: ListView.builder(
                    scrollDirection: Axis.horizontal,
                    itemCount: 10,

                    itemBuilder: (BuildContext context, int index) {
                      return Row(
                        children: [
                          SizedBox(
                            width: 10,
                          ),
                          Container(
                            decoration: BoxDecoration(
                                color: Color(0xff16161e),
                                borderRadius: BorderRadius.circular(10)),
                            width: 150,
                            child: Center(
                              child: Column(
                                mainAxisAlignment: MainAxisAlignment.center,
                                crossAxisAlignment: CrossAxisAlignment.start,
                                children: [
                                  Center(
                                    child: Text('Hello',
                                    style: TextStyle(fontSize: 30, fontWeight: FontWeight.w700,color: Colors.white),),
                                  )
                                ],
                              ),
                            ),
                          ),
                        ],
                      );
                    },
                  ),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }
}
```
3. To reproduce the bug tap on the nested Listview and any blank space at the same time. 
 

**Expected results:** Outer Listview to work normal.

**Actual results:** Outer Listview crashes.

  

<details>
  <summary>Logs</summary>

] flutter: ══╡ EXCEPTION CAUGHT BY GESTURE LIBRARY ╞═══════════════════════════════════════════════════════════
[   +7 ms] flutter: The following assertion was thrown while routing a pointer event:
[        ] flutter: 'package:flutter/src/gestures/monodrag.dart': Failed assertion: line 435 pos 12: '_initialButtons ==
[        ] flutter: kPrimaryButton': is not true.
[        ] flutter: 
[        ] flutter: Either the assertion indicates an error in the framework itself, or we should provide substantially
[        ] flutter: more information in this error message to help you determine and fix the underlying cause.
[        ] flutter: In either case, please report this assertion by filing a bug on GitHub:
[        ] flutter:   https://github.com/flutter/flutter/issues/new?template=BUG.md
[        ] flutter: 
[        ] flutter: When the exception was thrown, this was the stack:
[        ] flutter: #2      DragGestureRecognizer._checkEnd (package:flutter/src/gestures/monodrag.dart:435:12)
[        ] flutter: #3      DragGestureRecognizer.didStopTrackingLastPointer (package:flutter/src/gestures/monodrag.dart:376:9)
[  +12 ms] flutter: #4      OneSequenceGestureRecognizer.stopTrackingPointer (package:flutter/src/gestures/recognizer.dart:342:9)
[        ] flutter: #5      DragGestureRecognizer._giveUpPointer (package:flutter/src/gestures/monodrag.dart:385:5)
[        ] flutter: #6      DragGestureRecognizer.handleEvent (package:flutter/src/gestures/monodrag.dart:282:9)
[        ] flutter: #7      PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:77:12)
[        ] flutter: #8      PointerRouter._dispatchEventToRoutes.<anonymous closure> (package:flutter/src/gestures/pointer_router.dart:122:9)
[        ] flutter: #9      _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:377:8)
[        ] flutter: #10     PointerRouter._dispatchEventToRoutes (package:flutter/src/gestures/pointer_router.dart:120:18)
[        ] flutter: #11     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:106:7)
[        ] flutter: #12     GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:358:19)
[        ] flutter: #13     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:338:22)
[        ] flutter: #14     RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:267:11)
[        ] flutter: #15     GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:295:7)
[        ] flutter: #16     GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:240:7)
[        ] flutter: #17     GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:213:7)
[   +1 ms] flutter: #21     _invoke1 (dart:ui/hooks.dart:265:10)
[        ] flutter: #22     _dispatchPointerDataPacket (dart:ui/hooks.dart:174:5)
[   +1 ms] flutter: (elided 5 frames from class _AssertionError and dart:async)
[        ] flutter: 
[        ] flutter: router: Instance of 'PointerRouter'
[        ] flutter: route: Closure: (PointerEvent) => void from Function 'handleEvent':.
[        ] flutter: event: PointerMoveEvent#ed135(position: Offset(62.0, 506.5))
[        ] flutter: ════════════════════════════════════════════════════════════════════════════════════════════════════
[+6807 ms] flutter: Another exception was thrown: 'package:flutter/src/gestures/monodrag.dart': Failed assertion: line 435 pos 12:
'_initialButtons == kPrimaryButton': is not true.
[+26092 ms] flutter: Another exception was thrown: 'package:flutter/src/gestures/monodrag.dart': Failed assertion: line 435 pos 12:
'_initialButtons == kPrimaryButton': is not true.
[ +419 ms] flutter: Another exception was thrown: 'package:flutter/src/gestures/monodrag.dart': Failed assertion: line 435 pos 12:
'_initialButtons == kPrimaryButton': is not true.

```
```

[✓] Flutter (Channel stable, 1.22.6, on macOS 11.1 20C69 darwin-x64, locale en-IN)
    • Flutter version 1.22.6 at /Users/rahul/Desktop/flutter
    • Framework revision 9b2d32b605 (8 days ago), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for
      detailed instructions).
      If the Android SDK has been installed to a custom location, set ANDROID_SDK_ROOT
      to that location.
      You may also want to add it to your PATH environment variable.

 
[✓] Xcode - develop for iOS and macOS (Xcode 12.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.0

[!] Android Studio (not installed)
    • Android Studio not found; download from
      https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for
      detailed instructions).

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (1 available)
    • R’s I  (mobile) • c6467d3f454b96020193bf65ecad69cb68dbf640 • ios • iOS 14.3
</details>

### PR DESCRIPTION
**Pull request by goderbauer**
Fixes https://github.com/flutter/flutter/issues/75061.

https://github.com/flutter/flutter/issues/75061#issuecomment-778498362 contains a detailed description of what the problem was that's fixed by this PR.

### CODE REVIEW

**Comment by goderbauer** [comment]
@dkwingsmt @HansMuller @jason-simmons

File: packages/flutter/lib/src/gestures/monodrag.dart [hunk 1]
```diff
@@ -309,15 +309,16 @@ abstract class DragGestureRecognizer extends OneSequenceGestureRecognizer {
...
     if (event is PointerUpEvent || event is PointerCancelEvent) {
-      _giveUpPointer(
-        event.pointer,
-        reject: event is PointerCancelEvent || _state ==_DragState.possible,
-      );
+      _giveUpPointer(event.pointer);
     }
   }
 
+  final List<int> _acceptedActivePointers = <int>[];
```
Comment (dkwingsmt) [inline packages/flutter/lib/src/gestures/monodrag.dart:316]: According to the use cases, shouldn't `Set` be a better container here?

**Review by dkwingsmt — APPROVED** [review]
LGTM except for a small comment

File: packages/flutter/lib/src/gestures/monodrag.dart [hunk 2]
```diff
@@ -384,20 +385,23 @@ abstract class DragGestureRecognizer extends OneSequenceGestureRecognizer {
...
-      localPosition: _initialPosition.local,
-    );
-    if (onDown != null)
+    if (onDown != null) {
+      final DragDownDetails details = DragDownDetails(
+        globalPosition: _initialPosition.global,
+        localPosition: _initialPosition.local,
+      );
       invokeCallback<void>('onDown', () => onDown!(details));
+    }
```
Comment (xu-baolin) [inline packages/flutter/lib/src/gestures/monodrag.dart:404]: 👍 

`_checkStart` and `_checkUpdate` also can be optimized.

**Pull request merged**